hangman.py

```diff
@@ -1,213 +1,304 @@
+    '''
+    alphabet = string.ascii_lowercase
+    new_alphabet = []
+    for letter in alphabet:
+        if letter in letters_guessed:
+            new_alphabet += ''
+        else:
+            new_alphabet += f'{letter}'
+
+    return ''.join(new_alphabet)
+
+
+def victory(guesses: int, secret_word: str) -> str:
+    """
+    guesses: count of guesses left
+    secret_word: the word the user is guessing
+    returns: string with congratulations and total score
+    """
+    total_score = guesses * len(set(secret_word))
+    return f'\033[7mCongratulations! You won, the secret word was - {secret_word}\033[0m\n\033[7mYour total score is {total_score}\033[0m'
+
+
+def letter_check(guesses: int, warnings: int) -> tuple:
+    """
+    guesses: count of guesses left
+    warnings: count of warnings left
+    returns: which consist of letter, updated count of guesses, updated count of warnings
+    """
+    while 1:
+        letter = input('Enter your guess letter:   ')
+        if letter.isalpha():
+            break
+        else:
+            if warnings <= 0:
+                guesses -= 1
+                if guesses == 0:
+                    break
+            else:
+                warnings -= 1
+            print(
+                f'\033[31mYou must enter a letter of the Latin alphabet. Not a number or symbol. {warnings} warnings left\033[0m')
+    return letter, guesses, warnings
+
+
+def hangman(secret_word):
+    '''
+    secret_word: string, the secret word to guess.
+
+    Starts up an interactive game of Hangman.
+
+    * At the start of the game, let the user know how many
+      letters the secret_word contains and how many guesses s/he starts with.
+
+    * The user should start with 6 guesses
+    * Before each round, you should display to the user how many guesses
+      s/he has left and the letters that the user has not yet guessed.
+
+    * Ask the user to supply one guess per round. Remember to make
+      sure that the user puts in a letter!
+
+    * The user should receive feedback immediately after each guess
+      about whether their guess appears in the computer's word.
+    * After each guess, you should display to the user the
+      partially guessed word so far.
+
+    Follows the other limitations detailed in the problem write-up.
+    '''
+    print('''
+ _   _                                           __   _____ 
+| | | |                                         /  | |  _  |
+| |_| | __ _ _ __   __ _ _ __ ___   __ _ _ __   `| | | |/' |
+|  _  |/ _` | '_ \ / _` | '_ ` _ \ / _` | '_ \   | | |  /| |
+| | | | (_| | | | | (_| | | | | | | (_| | | | | _| |_\ |_/ /
+\_| |_/\__,_|_| |_|\__, |_| |_| |_|\__,_|_| |_| \___(_)___/ 
+                    __/ |                                   
+                   |___/                                    
+''')
+    print(f'I am thinking of a word that is {len(secret_word)} letters long.')
+    guesses = 6
+    letters_guessed = []
+    vowels = ('a', 'e', 'i', 'o', 'u')

```

Comment:
Why not? For vowels we need immutable container and for letters_guessed - mutable. So its tuple and list

Reply:
You need not only immutable container, but also such container where you can perform check for 'letter exists in container' quite fast. In case of tuple an list - Python will look at each letter and will try to compare it to your letter. There's another container exist in Python where it won't need to look at each letter.